src/etc/gdb_rust_pretty_printing.py

```diff
@@ -333,27 +333,37 @@ def children(self):
 # Yield each key (and optionally value) from a BoxedNode.
 def children_of_node(boxed_node, height, want_values):
     node_ptr = boxed_node['ptr']['pointer']
-    if height > 0:
-        type_name = str(node_ptr.type.target()).replace('LeafNode', 'InternalNode')
-        node_type = gdb.lookup_type(type_name)
-        node_ptr = node_ptr.cast(node_type.pointer())
-        leaf = node_ptr['data']
-    else:
-        leaf = node_ptr.dereference()
-    keys = leaf['keys']
-    if want_values:
-        values = leaf['vals']
-    length = int(leaf['len'])
-    for i in xrange(0, length + 1):
+    head = node_ptr.dereference()
+    length = int(head['len'])
+    if length > 0:
```

Comment:
This `if` and moving `length` may not have been necessary. I don't quite understand yet how it's called.

Reply:
moved off to #68098